### PR TITLE
Hide translation interface in notice/experience form when translations aren't enabled

### PR DESCRIPTION
**Pull request by jpople**
Closes PROD-1820

### Description Of Changes

Removes references to translations in the privacy notice and privacy experience forms when user does not have translations enabled on the backend, replacing them with generic "text editing" forms.

### Code Changes

* [ ] Small refactors to privacy notice form and privacy experience form
* [ ] Re-generate OpenAPI types

### Steps to Confirm

Use fidesplus backend on branch `PROD-1819_surface_translation_config_settings`.

As a control to confirm translation interfaces still work as expected when translations are enabled, have the following in `.fides/fides.toml` in fidesplus:
```toml
[consent]
enable_translations = true
enable_oob_translations = true
```
* [ ] In admin UI, navigate to `/consent/privacy-notices`
* [ ] Edit or create a new privacy notice
* [ ] Should see interface at bottom of form for adding new languages and switching between languages
* [ ] Should be able to make changes to translations and submit notices
* [ ] In admin UI, navigate to `/consent/privacy-experience`
* [ ] Edit or create a new privacy experience
* [ ] Should see scrollable list component with languages; should be able to add new translations and edit existing ones
* [ ] Should be able to make changes to translations and submit experience

Replace `true` with `false` in the config snippet above, then rerun backend.
* [ ] Navigate to `/consent/privacy-notices`
* [ ] Edit or create a new privacy notice
* [ ] Should see form titled "Notice text" and no language selection menu:

Screenshot 2024-03-19 at 11 14 33

* [ ] Should be able to make changes to notice text and submit notice
* [ ] Navigate to `/consent/privacy-experience`
* [ ] Edit or create a new privacy experience
* [ ] Should see button labeled "Edit experience text" and no language selection or "auto detect language" switch
* [ ] Inside text edit form, switch to toggle whether language is default should not be present
* [ ] Should be able to make changes to experience text and submit experience

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* [ ] Issue Requirements are Met
* [ ] Update `CHANGELOG.md`

### CODE REVIEW

File: clients/admin-ui/src/types/api/models/ExperienceConfigResponse.ts [hunk 1]
```diff
@@ -24,4 +25,5 @@ export type ExperienceConfigResponse = {
   component: ComponentType;
   privacy_notices?: Array<PrivacyNoticeResponse>;
   translations?: Array<ExperienceTranslationResponse>;
+  properties?: Array<MinimalProperty>;
```
Comment (eastandwestwind) [inline clients/admin-ui/src/types/api/models/ExperienceConfigResponse.ts:28]: 💯

File: clients/admin-ui/src/types/api/models/SupportedLanguage.ts [hunk 2]
```diff
@@ -16,6 +16,7 @@ export enum SupportedLanguage {
   EL = "el",
   EN = "en",
   ES = "es",
+  ES_MX = "es-MX",
```
Comment (eastandwestwind) [inline clients/admin-ui/src/types/api/models/SupportedLanguage.ts:19]: was this just something we missed before?

**Review by eastandwestwind — APPROVED** [review]
Looks great, and thanks for generating the types again 🌟

File: clients/admin-ui/src/types/api/models/SupportedLanguage.ts (hunk 2 above)
Comment (jpople) [reply clients/admin-ui/src/types/api/models/SupportedLanguage.ts:19]: I think there was some uncertainty with whether we would support Mexican Spanish in addition to Castillian, but I guess that got resolved?  Not 100% sure, to be honest.

Comment (pattisdr) [reply clients/admin-ui/src/types/api/models/SupportedLanguage.ts:19]: This was added later, we originally only had "es" (and we saving mexican spanish translations underneath them)! And then we received castillian translations so we switched them and those were saved under es, and we added new "es-mx" for Spanish (Mexico)!

**Pull request merged**